### DISCUSSION TITLE
How to get "colour-checker-detection" packaged with "Pyinstaller"?

### DISCUSSION THREAD

**Discussion by farhanalfin** [colour-science/colour-checker-detection#16] [Q&A]
I work for a project over scanner calibration , It works well from pycharm but when I convert the projet to exe using pyinsatller the exe shout down , to debuge the issue , I exculde  colour-checker-detection module, the exe work well.

**Comment by KelSolaar** [comment]
Hi @farhanalfin,

So freezing/packaging *colour-checker-detection* with Pyinstaller is not something I have done. Having done a lot of it with other projects orders of magnitude more complex than this one, I suspect that you are missing a lot of hidden imports, e.g. `colour`, `scipy`, `opencv`, etc... I would recommend using the console & debug modes and see what imports are missing!

Worth noting that @mrlixm packages `colour*` in his [ColorspaceConvert](https://mrlixm.github.io/PYCO/standalone/ColorspaceConvert/home/) tool, maybe he can give you some more info.

**Reply by MrLixm** [reply]
Hello, thanks for the ping Thomas.
USing the pyinstaller debug mode is a best way to see whats missing.
I had an issue with scipy that I explained [in the doc here](https://mrlixm.github.io/PYCO/standalone/ColorspaceConvert/development/#fix-scipy-lib-not-found-issue).

Hope this can help you.
Cheers.
Liam.

**Comment by KelSolaar** [comment]
@farhanalfin : I suppose you are not intending to use this package anymore considering this question: https://stackoverflow.com/questions/66302777/converting-color-correction-opencv-module-example-from-c-to-python

Can you please let us know so that I can close the discussion?

**Comment by farhanalfin** [comment]
Dear Thomas
thank you very much for your email.

I am still interested in using all possible modules involved in colour
checker detector and colour correction.
Actually after solving the problem I faced with OpenCV color correction
module I will compare between speed of OpencCv and colour model.

After I opened the discussion  How to get "colour-checker-detection"
packaged with "Pyinstaller"? (#16)
I tried to debug to solve the bug and I found that the problem is from
using numpy.linalg.pinv at
colour-science/colour/blob/develop/colour/algebra/regression.py
line 67 return np.dot(np.transpose(x), np.linalg.pinv(np.transpose(y)))

Actually till now I could not solve this problem.

Again thanks for your patience , I am new at Python, and thanks for helping
me.

Sincerely
Farhan ALFİN

**Reply by KelSolaar** [reply]
Can you share a traceback or anything?

**Comment by farhanalfin** [comment]
this is the simplest example of using numpy with pyinstaller

import numpy as np

from numpy.linalg import linalg

a = np.array([[1., 2.], [3., 4.]])

print(linalg.pinv)

print(linalg.pinv(a))


work very well from pycharm but after freezing by pyinstaller with the
simplest commande

pyinstaller sim_np.py

the exe exit without any traceback

**Reply by KelSolaar** [reply]
Just worked here on macOS:

```
(colour-package) Kali:Environments kelsolaar$ pyinstaller np.py
329 INFO: PyInstaller: 4.2
329 INFO: Python: 3.8.6
428 INFO: Platform: macOS-10.15.7-x86_64-i386-64bit
437 INFO: wrote /Users/kelsolaar/Documents/Development/Environments/np.spec
500 INFO: UPX is not available.
505 INFO: Extending PYTHONPATH with paths
['/Users/kelsolaar/Documents/Development/Environments',
 '/Users/kelsolaar/Documents/Development/Environments']
560 INFO: checking Analysis
560 INFO: Building Analysis because Analysis-00.toc is non existent
560 INFO: Initializing module dependency graph...
567 INFO: Caching module graph hooks...
594 WARNING: Several hooks defined for module 'win32ctypes.core'. Please take care they do not conflict.
705 INFO: Analyzing base_library.zip ...
24288 INFO: Processing pre-find module path hook distutils from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks/pre_find_module_path/hook-distutils.py'.
24298 INFO: distutils: retargeting to non-venv dir '/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8'
42295 INFO: Caching module dependency graph...
43346 INFO: running Analysis Analysis-00.toc
43464 INFO: Analyzing /Users/kelsolaar/Documents/Development/Environments/np.py
49042 INFO: Processing pre-find module path hook site from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks/pre_find_module_path/hook-site.py'.
49047 INFO: site: retargeting to fake-dir '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/fake-modules'
61738 INFO: Processing module hooks...
61740 INFO: Loading module hook 'hook-pkg_resources.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
63230 INFO: Processing pre-safe import module hook win32com from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/_pyinstaller_hooks_contrib/hooks/pre_safe_import_module/hook-win32com.py'.
63297 WARNING: Hidden import "pkg_resources.py2_warn" not found!
63297 WARNING: Hidden import "pkg_resources.markers" not found!
63301 INFO: Excluding import of __main__ from module pkg_resources
63302 INFO: Loading module hook 'hook-scipy.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
63303 INFO: Loading module hook 'hook-xml.etree.cElementTree.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
63309 INFO: Loading module hook 'hook-lib2to3.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
63563 INFO: Loading module hook 'hook-_tkinter.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
63573 INFO: Not collecting Tcl/Tk data - either python is using macOS' system Tcl/Tk framework, or Tcl/Tk data directories could not be found.
63573 INFO: Loading module hook 'hook-encodings.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
63750 INFO: Loading module hook 'hook-distutils.util.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
63762 INFO: Excluding import of lib2to3.refactor from module distutils.util
63764 INFO: Loading module hook 'hook-setuptools.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
65947 INFO: Import to be excluded not found: 'setuptools.py33compat'
65947 INFO: Import to be excluded not found: 'setuptools.py27compat'
65947 INFO: Loading module hook 'hook-pickle.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
65953 INFO: Excluding import of argparse from module pickle
65953 INFO: Loading module hook 'hook-heapq.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
65957 INFO: Excluding import of doctest from module heapq
65958 INFO: Loading module hook 'hook-difflib.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
65963 INFO: Excluding import of doctest from module difflib
65963 INFO: Loading module hook 'hook-multiprocessing.util.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
65970 INFO: Excluding import of test from module multiprocessing.util
65970 INFO: Excluding import of test.support from module multiprocessing.util
65970 INFO: Loading module hook 'hook-sysconfig.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
66124 INFO: Loading module hook 'hook-numpy.core.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
73507 INFO: Loading module hook 'hook-xml.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
73612 INFO: Loading module hook 'hook-numpy.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
73616 INFO: Loading module hook 'hook-distutils.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
73618 INFO: Loading module hook 'hook-numpy._pytesttester.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
73671 INFO: Excluding import of pytest from module numpy._pytesttester
73672 INFO: Loading module hook 'hook-setuptools.msvc.py' from '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks'...
73678 INFO: Excluding import of numpy from module setuptools.msvc
73857 INFO: Looking for ctypes DLLs
74215 INFO: Analyzing run-time hooks ...
74243 INFO: Including run-time hook '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks/rthooks/pyi_rth_multiprocessing.py'
74260 INFO: Including run-time hook '/Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/hooks/rthooks/pyi_rth_pkgres.py'
74457 INFO: Looking for dynamic libraries
75029 INFO: Looking for eggs
75029 INFO: Using Python library /usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/Python
75109 INFO: Warnings written to /Users/kelsolaar/Documents/Development/Environments/build/np/warn-np.txt
75547 INFO: Graph cross-reference written to /Users/kelsolaar/Documents/Development/Environments/build/np/xref-np.html
75724 INFO: checking PYZ
75724 INFO: Building PYZ because PYZ-00.toc is non existent
75724 INFO: Building PYZ (ZlibArchive) /Users/kelsolaar/Documents/Development/Environments/build/np/PYZ-00.pyz
78941 INFO: Building PYZ (ZlibArchive) /Users/kelsolaar/Documents/Development/Environments/build/np/PYZ-00.pyz completed successfully.
79005 INFO: checking PKG
79005 INFO: Building PKG because PKG-00.toc is non existent
79005 INFO: Building PKG (CArchive) PKG-00.pkg
79075 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
79081 INFO: Bootloader /Users/kelsolaar/Documents/Development/Environments/colour-package/lib/python3.8/site-packages/PyInstaller/bootloader/Darwin-64bit/run
79082 INFO: checking EXE
79084 INFO: Building EXE because EXE-00.toc is non existent
79084 INFO: Building EXE from EXE-00.toc
79085 INFO: Appending archive to EXE /Users/kelsolaar/Documents/Development/Environments/build/np/np
79109 INFO: Fixing EXE for code signing /Users/kelsolaar/Documents/Development/Environments/build/np/np
79116 INFO: Building EXE from EXE-00.toc completed successfully.
79119 INFO: checking COLLECT
79119 INFO: Building COLLECT because COLLECT-00.toc is non existent
79120 INFO: Building COLLECT COLLECT-00.toc
83197 INFO: Building COLLECT COLLECT-00.toc completed successfully.
(colour-package) Kali:Environments kelsolaar$ dist/np/./np
<function pinv at 0x10d5fc310>
[[-2.   1. ]
 [ 1.5 -0.5]]
```

Which OS are you on?

**Comment by farhanalfin** [comment]
My system is windows 10 64 bit
python 3.9 32 bit

(base) PS C:\Users\newge\pycharmprojects\flour_color> pyinstaller sim_np.py
10732 INFO: PyInstaller: 4.2
10732 INFO: Python: 3.8.5 (conda)
10732 INFO: Platform: Windows-10-10.0.19041-SP0
10732 INFO: wrote C:\Users\newge\pycharmprojects\flour_color\sim_np.spec
10732 INFO: UPX is not available.
10748 INFO: Extending PYTHONPATH with paths
['C:\\Users\\newge\\pycharmprojects\\flour_color',
 'C:\\Users\\newge\\pycharmprojects\\flour_color']
10857 INFO: checking Analysis
10857 INFO: Building Analysis because Analysis-00.toc is non existent
10857 INFO: Initializing module dependency graph...
10873 INFO: Caching module graph hooks...
10873 WARNING: Several hooks defined for module 'win32ctypes.core'. Please
take care they do not conflict.
10888 INFO: Analyzing base_library.zip ...
16283 INFO: Processing pre-find module path hook distutils from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-distutils.py'.
16283 INFO: distutils: retargeting to non-venv dir
'c:\\programdata\\anaconda3\\lib'
20989 INFO: Caching module dependency graph...
21235 INFO: running Analysis Analysis-00.toc
21258 INFO: Adding Microsoft.Windows.Common-Controls to dependent
assemblies of final executable
  required by c:\programdata\anaconda3\python.exe
21675 INFO: Analyzing C:\Users\newge\pycharmprojects\flour_color\sim_np.py
25474 INFO: Processing pre-find module path hook site from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-site.py'.
25476 INFO: site: retargeting to fake-dir
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\fake-modules'
35233 INFO: Processing pre-safe import module hook six.moves from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\
hook-six.moves.py'.
41555 INFO: Processing module hooks...
41556 INFO: Loading module hook 'hook-pycparser.py' from
'C:\\Users\\newge\\AppData\\Roaming\\Python\\Python38\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
41560 INFO: Loading module hook 'hook-pytest.py' from
'C:\\Users\\newge\\AppData\\Roaming\\Python\\Python38\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
43451 INFO: Loading module hook 'hook-win32ctypes.core.py' from
'C:\\Users\\newge\\AppData\\Roaming\\Python\\Python38\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
43816 INFO: Loading module hook 'hook-difflib.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
43819 INFO: Excluding import of doctest from module difflib
43820 INFO: Loading module hook 'hook-distutils.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
43823 INFO: Loading module hook 'hook-distutils.util.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
43826 INFO: Excluding import of lib2to3.refactor from module distutils.util
43827 INFO: Loading module hook 'hook-encodings.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
43941 INFO: Loading module hook 'hook-heapq.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
43964 INFO: Excluding import of doctest from module heapq
43965 INFO: Loading module hook 'hook-importlib_metadata.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
43972 INFO: Loading module hook 'hook-lib2to3.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
44061 INFO: Loading module hook 'hook-multiprocessing.util.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
44066 INFO: Excluding import of test.support from module
multiprocessing.util
44066 INFO: Excluding import of test from module multiprocessing.util
44067 INFO: Loading module hook 'hook-numpy.core.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
44744 INFO: MKL libraries found when importing numpy. Adding MKL to binaries
44750 INFO: Loading module hook 'hook-numpy.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
44752 INFO: Loading module hook 'hook-numpy._pytesttester.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
44756 INFO: Excluding import of pytest from module numpy._pytesttester
44756 INFO: Loading module hook 'hook-packaging.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
44758 INFO: Loading module hook 'hook-pickle.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
44761 INFO: Excluding import of argparse from module pickle
44761 INFO: Loading module hook 'hook-pkg_resources.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
46065 INFO: Processing pre-safe import module hook win32com from
'C:\\Users\\newge\\AppData\\Roaming\\Python\\Python38\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\pre_safe_import_module\\hook-win32com.py'.
46549 WARNING: Hidden import "pkg_resources.py2_warn" not found!
46569 WARNING: Hidden import "pkg_resources.markers" not found!
46571 INFO: Excluding import of __main__ from module pkg_resources
46572 INFO: Loading module hook 'hook-pygments.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
49582 INFO: Loading module hook 'hook-PyQt5.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
50344 WARNING: Hidden import "PyQt5.sip" not found!
50345 INFO: Loading module hook 'hook-PyQt5.QtCore.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
50438 INFO: Loading module hook 'hook-PyQt5.QtGui.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
50594 INFO: Loading module hook 'hook-scipy.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
50599 INFO: Loading module hook 'hook-setuptools.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
51961 INFO: Import to be excluded not found: 'setuptools.py27compat'
51961 INFO: Import to be excluded not found: 'setuptools.py33compat'
51962 INFO: Loading module hook 'hook-sysconfig.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
51967 INFO: Loading module hook 'hook-xml.dom.domreg.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
51968 INFO: Loading module hook 'hook-xml.etree.cElementTree.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
51969 INFO: Loading module hook 'hook-xml.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
51970 INFO: Loading module hook 'hook-zope.interface.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
51974 INFO: Excluding import of unittest from module zope.interface._compat
51974 INFO: Loading module hook 'hook-_tkinter.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
52395 INFO: checking Tree
52396 INFO: Building Tree because Tree-00.toc is non existent
52397 INFO: Building Tree Tree-00.toc
52658 INFO: checking Tree
52659 INFO: Building Tree because Tree-01.toc is non existent
52660 INFO: Building Tree Tree-01.toc
52798 INFO: checking Tree
52798 INFO: Building Tree because Tree-02.toc is non existent
52799 INFO: Building Tree Tree-02.toc
52810 INFO: Loading module hook 'hook-pythoncom.py' from
'C:\\Users\\newge\\AppData\\Roaming\\Python\\Python38\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
53561 INFO: Loading module hook 'hook-pywintypes.py' from
'C:\\Users\\newge\\AppData\\Roaming\\Python\\Python38\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
54341 INFO: Loading module hook 'hook-win32com.py' from
'C:\\Users\\newge\\AppData\\Roaming\\Python\\Python38\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
c:\programdata\anaconda3\lib\site-packages\win32com\client\makepy.py:369:
SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if path is not '' and not os.path.exists(path):
55381 INFO: Loading module hook 'hook-PIL.Image.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
55997 INFO: Loading module hook 'hook-PIL.ImageFilter.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
56001 INFO: Excluding import of numpy from module PIL.ImageFilter
56001 INFO: Loading module hook 'hook-PIL.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
56008 INFO: Excluding import of PyQt5.QtGui from module PIL.ImageQt
56008 INFO: Excluding import of PyQt5.QtCore from module PIL.ImageQt
56010 INFO: Import to be excluded not found: 'FixTk'
56013 INFO: Excluding import of tkinter from module PIL.ImageTk
56013 INFO: Loading module hook 'hook-PIL.SpiderImagePlugin.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
56018 INFO: Import to be excluded not found: 'FixTk'
56020 INFO: Loading module hook 'hook-setuptools.msvc.py' from
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks'...
56024 INFO: Excluding import of numpy from module setuptools.msvc
56080 INFO: Looking for ctypes DLLs
56232 INFO: Analyzing run-time hooks ...
56243 INFO: Including run-time hook
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_multiprocessing.py'
56248 INFO: Including run-time hook
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_win32api.py'
56249 INFO: Including run-time hook
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_pkgres.py'
56252 INFO: Including run-time hook
'c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_win32comgenpy.py'
56283 INFO: Looking for dynamic libraries
57897 INFO: Looking for eggs
57897 INFO: Using Python library c:\programdata\anaconda3\python38.dll
57899 INFO: Found binding redirects:
[]
57913 INFO: Warnings written to
C:\Users\newge\pycharmprojects\flour_color\build\sim_np\warn-sim_np.txt
58152 INFO: Graph cross-reference written to
C:\Users\newge\pycharmprojects\flour_color\build\sim_np\xref-sim_np.html
58201 INFO: checking PYZ
58201 INFO: Building PYZ because PYZ-00.toc is non existent
58202 INFO: Building PYZ (ZlibArchive)
C:\Users\newge\pycharmprojects\flour_color\build\sim_np\PYZ-00.pyz
61072 INFO: Building PYZ (ZlibArchive)
C:\Users\newge\pycharmprojects\flour_color\build\sim_np\PYZ-00.pyz
completed successfully.
61116 INFO: checking PKG
61116 INFO: Building PKG because PKG-00.toc is non existent
61117 INFO: Building PKG (CArchive) PKG-00.pkg
61254 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
61257 INFO: Bootloader
c:\programdata\anaconda3\lib\site-packages\PyInstaller\bootloader\Windows-32bit\run.exe
61257 INFO: checking EXE
61258 INFO: Building EXE because EXE-00.toc is non existent
61258 INFO: Building EXE from EXE-00.toc
61274 INFO: Copying icons from
['c:\\programdata\\anaconda3\\lib\\site-packages\\PyInstaller\\bootloader\\images\\icon-console.ico']
61315 INFO: Writing RT_GROUP_ICON 0 resource with 104 bytes
61316 INFO: Writing RT_ICON 1 resource with 3752 bytes
61317 INFO: Writing RT_ICON 2 resource with 2216 bytes
61318 INFO: Writing RT_ICON 3 resource with 1384 bytes
61318 INFO: Writing RT_ICON 4 resource with 37019 bytes
61319 INFO: Writing RT_ICON 5 resource with 9640 bytes
61320 INFO: Writing RT_ICON 6 resource with 4264 bytes
61320 INFO: Writing RT_ICON 7 resource with 1128 bytes
61353 INFO: Appending archive to EXE
C:\Users\newge\pycharmprojects\flour_color\build\sim_np\sim_np.exe
61521 INFO: Building EXE from EXE-00.toc completed successfully.
61524 INFO: checking COLLECT
61524 INFO: Building COLLECT because COLLECT-00.toc is non existent
61525 INFO: Building COLLECT COLLECT-00.toc
74818 INFO: Building COLLECT COLLECT-00.toc completed successfully.


when executing
C:\Users\newge\PycharmProjects\Flour_Color\dist\sim_np>sim_np
<function pinv at 0x038470B8>

C:\Users\newge\PycharmProjects\Flour_Color\dist\sim_np>
